src/LibraryUtilities.ts

```diff
@@ -566,48 +285,244 @@ export function updateSections(oldLayoutSpecs: any, newLayoutSpecs: any): void {
 }
 
 /**
- * Convert an array of typeListNodes to ItemData which are not processed based on their fullyQualifiedNames, 
- * by splitting the name with '.', and put them under a specific section.
+ * This function will convert layoutElements to ItemData, and construct an array of IncludeItemPair,
+ * which stores an includeInfo and its parent item.
  * 
- * As an example, for two typeListNodes 'A.B.C' and 'A.B.D', the resulting nodes are:
- * - A
- *   |- B
+ * @param {LayoutElement[]} layoutElements 
+ * The layoutElements to be converted to ItemData
+ * 
+ * @param {IncludeItemPair[]} pairs
+ * The IncludeItemPairs to be contructed from
+ * 
+ * @param {ItemData} parentItem 
+ * The parentItem of layoutElements (if exist)
+ */
+function convertLayoutElementToItemData(
+    layoutElements: LayoutElement[],
+    pairs: IncludeItemPair[],
+    parentItem?: ItemData):
+    ItemData[] {
+
+    let results: ItemData[] = [];
+
+    for (let layoutElement of layoutElements) {
+        let layoutData = new ItemData(layoutElement.text);
+        layoutData.constructFromLayoutElement(layoutElement);
+
+        // Sections that don't show its header will be expanded by default.
+        if (layoutData.itemType === "section" && !layoutData.showHeader) {
+            layoutData.expanded = true;
+        }
+
+        if (parentItem) {
+            parentItem.appendChild(layoutData);
+        }
+
+        if (layoutElement.include.length > 0) {
+            for (let include of layoutElement.include) {
+                let pair = new IncludeItemPair();
+                pair.include = include;
+                pair.parentItem = layoutData;
+                pairs.push(pair);
```

Comment:
Let's update constructor of `IncludeItemPair` so that the following is possible:

```js
pairs.push(new IncludeItemPair(include, layoutData));
```